tdp/core/models/deployment_log.py

```diff
@@ -226,6 +227,61 @@ def from_operations(
                 i += 1
         return deployment_log
 
+    @staticmethod
+    def from_operations_hosts_vars(
+        collections: Collections,
+        operation_host_vars_names: list[tuple[str, Optional[str], Optional[list[str]]]],
+    ) -> "DeploymentLog":
+        """Generate a deployment plan from a list of operations, hosts and extra vars.
+
+        Args:
+            collections: Collections to retrieve the operations from.
+            operation_host_vars_names: List of operations, hosts and extra vars to perform.
+
+        Raises:
+            MissingOperationError: If an operation is a noop.
+            MissingHostForOperationError: If an operation is not found in the collections.
+        """
+        (operation_names, host_names, var_names) = [
+            list(item) for item in zip(*operation_host_vars_names)
+        ]
+        collections.check_operations_exist(operation_names)
+
+        #   -  hosts_list : list of hosts, sorted and without duplicates ; if an operation is executed on eveyr host, the result is None (all the hosts)
+        #   -  extra_vars_list : list of extra vars (which is list of lists), sorted and without duplicates
```

Comment:
Not a correct way to comment the following vars. It is more like [inline comments](https://peps.python.org/pep-0008/#inline-comments).

Reply:
I prefered removing these comments : they didn't add anything to the code